gerritlab/main.py

```diff
@@ -130,15 +130,12 @@ def cancel_prev_pipelines(commit):
         commits_data.append(Commit(c, source_branch, target_branch))
 
     # Workflow:
-    # 1) Single push with all branch updates.
-    # 2) Create or update MRs.
+    # 1) Create or update MRs.
+    # 2) Single push with all branch updates.
+    # This order works best with GitLab.  If the order is swapped GitLab
+    # can become confused (xref https://gitlab.com/gitlab-org/gitlab-foss/-/issues/368).
 
-    # Push commits to Change-Id-named branches
-    refs_to_push = ["{}:refs/heads/{}".format(
-        c.commit.hexsha, c.source_branch) for c in commits_data]
-    with timing("push"):
-        remote.push(refspec=refs_to_push, force=True)        
-    
+    #print("Looking for existing MRs")
```

Comment:
Remove the debug prints before merging this.

Reply:
Done.